graphql/resolvers.go

```diff
@@ -831,6 +831,74 @@ func (r *mutationResolver) RestartTask(ctx context.Context, taskID string) (*res
 	return apiTask, err
 }
 
+func (r *mutationResolver) SaveSubscription(ctx context.Context, subscription restModel.APISubscription) (bool, error) {
+	usr := route.MustHaveUser(ctx)
+	username := usr.Username()
+
+	var id string
+	var idType string
+	for _, s := range subscription.Selectors {
```

Comment:
I would do a switch case here since there are many more types than this, many of which you don't need to handle yet: https://github.com/evergreen-ci/evergreen/blob/master/model/event/subscriptions.go#L126. This was part of the reason I said this probably doesn't need validation :)

Also handle type being null as well